integrations/gen_docs_integrations.py

```diff
@@ -228,49 +221,129 @@ def clean_and_write(md, txt):
+            if len(os.listdir(f'{path}/integrations')) == 1:
+                symlink_dict.update(
+                    {path: f'integrations/{clean_string(sidebar_label)}.md'})
             else:
-                # add custom_edit_url as the md file, so we can have uniqueness in the ingest script
-                # afterwards the ingest will replace this metadata with meta_yaml
-                md = md.replace("<!--startmeta",
-                                f'<!--startmeta\ncustom_edit_url: \"{meta_yaml.replace("metadata.yaml", "README.md")}')
-                finalpath = f'{path}/README.md'
-            try:
-                with open(finalpath, 'w') as txt:
-                    clean_and_write(md, txt)
-            except Exception as e:
-                print("Exception in notification md construction", e, integration['id'])
+                try:
+                    symlink_dict.pop(path)
+                except KeyError:
+                    # We don't need to print something here.
+                    pass
+    elif mode == 'notification':
 
-        except Exception as e:
-            print("Exception in for loop", e, "\n", integration)
-
-for element in symlink_dict:
-    # Remove the README to prevent it being a normal file
-    os.remove(f'{element}/README.md')
-    # and then make a symlink to the actual markdown
-    os.symlink(symlink_dict[element], f'{element}/README.md')
-
-    with open(f'{element}/{symlink_dict[element]}', 'r') as txt:
-        md = txt.read()
-
-    # This preserves the custom_edit_url for most files as it was,
-    # so the existing links don't break, this is vital for link replacement afterwards
-    with open(f'{element}/{symlink_dict[element]}', 'w+') as txt:
-        md = md.replace(f'{element}/{symlink_dict[element]}', f'{element}/README.md')
-        txt.write(md)
+        if "cloud-notifications" in path:
+            # for cloud notifications we generate them near their metadata.yaml
+            name = clean_string(integration['meta']['name'])
+
+            if not os.path.exists(f'{path}/integrations'):
+                os.mkdir(f'{path}/integrations')
+
+            # proper_edit_name = meta_yaml.replace(
+            #     "metadata.yaml", f'integrations/{clean_string(sidebar_label)}.md\"')
+
+            md = add_custom_edit_url(md, meta_yaml, sidebar_label, mode='cloud-notifications')
+
+            finalpath = f'{path}/integrations/{name}.md'
+        else:
+            # add custom_edit_url as the md file, so we can have uniqueness in the ingest script
+            # afterwards the ingest will replace this metadata with meta_yaml
+            md = add_custom_edit_url(md, meta_yaml, sidebar_label, mode='agent-notifications')
+
+            finalpath = f'{path}/README.md'
+        try:
+            with open(finalpath, 'w') as txt:
+                clean_and_write(md, txt)
+        except:
```

Comment:
Should probably be `OSError` here. As a general rule, if you’re just expecting errors related to file access or IO, that’s usually the correct thing to watch for unless you’re using something that provides more specific exceptions.